services/QuillLMS/app/services/skills_report/question_responses_for_classroom_by_skill.rb

```diff
@@ -51,12 +51,20 @@ class QuestionResponsesForClassroomBySkill < ApplicationService
             cr.activity_session_completed_at,
             #{skill_id_select} AS skill_id,
             #{skill_name_select} AS skill_name,
+            c_parent.id AS l1_skill_id,
+            c_parent.name AS l1_skill_name,
+            c_grandparent.id AS l2_skill_id,
+            c_grandparent.name AS l2_skill_name,
             ROUND(act_s.timespent::numeric / act.question_count, 0)::int AS timespent,
-            act_s.activity_id
+            act_s.activity_id,
+            (c_grandparent.gen_ed AND c_parent.gen_ed) AS gen_ed,
+            (c_grandparent.ell AND c_parent.ell) AS ell,
+            cu.classroom_id
         FROM students_classrooms sc
             JOIN activity_sessions act_s ON sc.student_id = act_s.user_id
-                #{'AND act_s.classroom_unit_id IN (SELECT id FROM classroom_unit_ids)' unless all_history}
+                #{'AND act_s.classroom_unit_id IN (SELECT id FROM classroom_unit_ids)' unless all_history || for_cache}
```

Comment:
Optional style thing: I find SQL is more readable and maintainable when Ruby logic is maximally extracted. How about:
```ruby
def join_classroom_unit_ids 
  return if all_history || for_cache
  'AND act_s.classroom_unit_id IN (SELECT id FROM classroom_unit_ids)'
end
```